--- origsrc/e2fsprogs-1.40.2/lib/ext2fs/ext2_fs.h	2007-06-30 07:58:34.000000000 -0500
+++ src/e2fsprogs-1.40.2/lib/ext2fs/ext2_fs.h	2007-08-20 23:50:32.093750000 -0500
@@ -447,6 +447,10 @@
 #define inode_uid(inode)	((inode).i_uid)
 #define inode_gid(inode)	((inode).i_gid)
 
+#else
+#define inode_uid(inode)	((inode).i_uid)
+#define inode_gid(inode)	((inode).i_gid)
+
 #endif  /* __masix__ */
 #endif  /* __GNU__ */
 #endif	/* defined(__KERNEL__) || defined(__linux__) */
--- origsrc/e2fsprogs-1.40.2/lib/ext2fs/getsize.c	2007-06-30 07:58:34.000000000 -0500
+++ src/e2fsprogs-1.40.2/lib/ext2fs/getsize.c	2007-08-20 23:46:48.328125000 -0500
@@ -61,7 +61,7 @@
 #include "ext2_fs.h"
 #include "ext2fs.h"
 
-#if defined(__CYGWIN__) || defined (WIN32)
+#if defined (WIN32) && !defined(__CYGWIN__)
 #include "windows.h"
 #include "winioctl.h"
 
--- origsrc/e2fsprogs-1.40.2/misc/mke2fs.c	2007-07-07 22:16:39.000000000 -0500
+++ src/e2fsprogs-1.40.2/misc/mke2fs.c	2007-08-21 00:01:39.500000000 -0500
@@ -495,11 +495,15 @@
 		}
 		uid = getuid();
 		inode.i_uid = uid;
+#ifdef i_uid_high
 		inode.i_uid_high = uid >> 16;
+#endif
 		if (uid) {
 			gid = getgid();
 			inode.i_gid = gid;
+#ifdef i_uid_high
 			inode.i_gid_high = gid >> 16;
+#endif
 		}
 		retval = ext2fs_write_new_inode(fs, EXT2_ROOT_INO, &inode);
 		if (retval) {
--- origsrc/e2fsprogs-1.40.2/resize/online.c	2007-06-30 07:58:35.000000000 -0500
+++ src/e2fsprogs-1.40.2/resize/online.c	2007-08-21 00:12:41.609375000 -0500
@@ -9,6 +9,10 @@
  * %End-Header%
  */
 
+#ifdef __CYGWIN__
+#define __USE_LINUX_IOCTL_DEFS
+#endif
+
 #include "resize2fs.h"
 #ifdef HAVE_SYS_IOCTL_H
 #include <sys/ioctl.h>
